formidling/src/main/java/no/nav/ung/sak/formidling/FormidlingTjeneste.java

```diff
@@ -0,0 +1,131 @@
+
+    private static final Logger LOG = LoggerFactory.getLogger(FormidlingTjeneste.class);
+
+    @Inject
+    public FormidlingTjeneste(
+        BrevGenerererTjeneste brevGenerererTjeneste,
+        VedtaksbrevRegler vedtaksbrevRegler,
+        VedtaksbrevValgRepository vedtaksbrevValgRepository,
+        BehandlingRepository behandlingRepository) {
+        this.brevGenerererTjeneste = brevGenerererTjeneste;
+        this.vedtaksbrevRegler = vedtaksbrevRegler;
+        this.vedtaksbrevValgRepository = vedtaksbrevValgRepository;
+        this.behandlingRepository = behandlingRepository;
+    }
+
+    FormidlingTjeneste() {
+    }
+
+
+    public VedtaksbrevValgDto vedtaksbrevValg(Long behandlingId) {
+        var behandling = behandlingRepository.hentBehandling(behandlingId);
+        var erAvsluttet = behandling.erAvsluttet();
+
+        var valg = vedtaksbrevValgRepository.finnVedtakbrevValg(behandlingId);
+
+        VedtaksbrevRegelResulat resultat = vedtaksbrevRegler.kjør(behandlingId);
+        LOG.info("VedtaksbrevRegelResultat: {}", resultat.safePrint());
+
+        var egenskaper = resultat.vedtaksbrevEgenskaper();
+
+        return new VedtaksbrevValgDto(
+            egenskaper.harBrev(),
+            null,
+            false,
+            egenskaper.kanHindre(),
+            valg.map(VedtaksbrevValgEntitet::isHindret).orElse(false),
+            !erAvsluttet && egenskaper.kanOverstyreHindre(),
+            egenskaper.kanRedigere(),
+            valg.map(VedtaksbrevValgEntitet::isRedigert).orElse(false),
+            !erAvsluttet && egenskaper.kanOverstyreRediger(),
+            resultat.forklaring(),
+            valg.map(VedtaksbrevValgEntitet::getRedigertBrevHtml).orElse(null)
+        );
+    }
+
+    public boolean måSkriveBrev(Long behandlingId) {
+        var regelResulat = vedtaksbrevRegler.kjør(behandlingId).vedtaksbrevEgenskaper();
+        return regelResulat.harBrev() && regelResulat.kanRedigere() && !regelResulat.kanOverstyreRediger();
+    }
+
+    public VedtaksbrevValgEntitet lagreVedtaksbrev(VedtaksbrevValgRequestDto dto) {
+        var behandling = behandlingRepository.hentBehandling(dto.behandlingId());
+        if (behandling.erAvsluttet()) {
+            throw new BadRequestException("Kan ikke endre vedtaksbrev på avsluttet behandling");
+        }
+
+        var vedtaksbrevValgEntitet = vedtaksbrevValgRepository.finnVedtakbrevValg(dto.behandlingId())
+            .orElse(VedtaksbrevValgEntitet.ny(dto.behandlingId()));
+
+        VedtaksbrevRegelResulat resultat = vedtaksbrevRegler.kjør(dto.behandlingId());
+        var vedtaksbrevEgenskaper = resultat.vedtaksbrevEgenskaper();
+
+        if (!vedtaksbrevEgenskaper.kanRedigere() && dto.redigert() != null) {
+            throw new BadRequestException("Brevet kan ikke redigeres.");
+        }
+
+        if (Boolean.TRUE.equals(dto.redigert()) && (dto.redigertHtml() == null || dto.redigertHtml().isBlank())) {
```

Comment:
Dette kan skrives som ein assert i dtoen

Reply:
Tenker du konstuktøren? Det er sjekk på to felter.

Reply:
men hvordan blir feilmeldingen ut?

Reply:
ok fant ut av det.